common/txmgr/confirmer.go

```diff
@@ -294,178 +260,156 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Pro
-	}
-
-	_, latestFinalizedHead, err := ec.headTracker.LatestAndFinalizedBlock(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to retrieve latest finalized head: %w", err)
-	}
-
-	if !latestFinalizedHead.IsValid() {
-		return fmt.Errorf("latest finalized head is not valid")
-	}
-
-	if latestFinalizedHead.BlockNumber() > head.BlockNumber() {
-		ec.lggr.Debugw("processHead received old block", "latestFinalizedHead", latestFinalizedHead.BlockNumber(), "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
+		return err
 	}
+	ec.lggr.Debugw("Finished SetBroadcastBeforeBlockNum", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
 
-	if err := ec.CheckForReceipts(ctx, head.BlockNumber(), latestFinalizedHead.BlockNumber()); err != nil {
-		return fmt.Errorf("CheckForReceipts failed: %w", err)
+	mark = time.Now()
+	if err := ec.CheckForConfirmation(ctx, head); err != nil {
+		return err
 	}
+	ec.lggr.Debugw("Finished CheckForConfirmation", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
 
-	ec.lggr.Debugw("Finished CheckForReceipts", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
 	mark = time.Now()
-
 	if err := ec.ProcessStuckTransactions(ctx, head.BlockNumber()); err != nil {
-		return fmt.Errorf("ProcessStuckTransactions failed: %w", err)
+		return err
 	}
-
 	ec.lggr.Debugw("Finished ProcessStuckTransactions", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
-	mark = time.Now()
 
+	mark = time.Now()
 	if err := ec.RebroadcastWhereNecessary(ctx, head.BlockNumber()); err != nil {
-		return fmt.Errorf("RebroadcastWhereNecessary failed: %w", err)
+		return err
 	}
-
 	ec.lggr.Debugw("Finished RebroadcastWhereNecessary", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
-	mark = time.Now()
-
-	if err := ec.EnsureConfirmedTransactionsInLongestChain(ctx, head, latestFinalizedHead.BlockNumber()); err != nil {
-		return fmt.Errorf("EnsureConfirmedTransactionsInLongestChain failed: %w", err)
-	}
+	ec.lggr.Debugw("processHead finish", "headNum", head.BlockNumber())
 
-	ec.lggr.Debugw("Finished EnsureConfirmedTransactionsInLongestChain", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
+	return nil
+}
 
-	if ec.resumeCallback != nil {
-		mark = time.Now()
-		if err := ec.ResumePendingTaskRuns(ctx, head); err != nil {
-			return fmt.Errorf("ResumePendingTaskRuns failed: %w", err)
+// CheckForConfirmation fetches the mined transaction count for each enabled address and marks transactions with a lower sequence as confirmed and ones with equal or higher sequence as unconfirmed
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CheckForConfirmation(ctx context.Context, head types.Head[BLOCK_HASH]) error {
+	for _, fromAddress := range ec.enabledAddresses {
+		// Returns the total transaction count for from address which is the highest mined sequence + 1
```

Comment:
nit: Let's drop this comment. This RPC call is straightforward, and the functionality is already described at the top.

Reply:
Will do, just added it in there because `SequenceAt` is sort of a misnomer